packages/apps/esm-offline-tools-app/src/offline-tools-page/offline-tools-page.component.tsx

```diff
@@ -0,0 +1,38 @@
+import { ExtensionSlot, useExtensionSlotMeta } from "@openmrs/esm-framework";
+import React from "react";
+import { RouteComponentProps, useRouteMatch } from "react-router-dom";
+import { OfflineToolsPageConfig } from "../types";
+import trimEnd from "lodash-es/trimEnd";
+
+export interface OfflineToolsPageParams {
+  page: string;
+}
+
+const OfflineToolsPage: React.FC<RouteComponentProps<OfflineToolsPageParams>> =
+  ({ match }) => {
+    const basePath = trimEnd(window.getOpenmrsSpaBase(), "/") + match.url;
+    const page = match.params.page;
+    const meta = useExtensionSlotMeta<OfflineToolsPageConfig>(
+      "offline-tools-page-slot"
+    );
+    const pageConfig = Object.values(meta).find(
+      (pageConfig) => pageConfig.name === page
+    );
+
+    if (!pageConfig) {
+      return null;
+    }
+
+    return (
+      <>
+        <ExtensionSlot extensionSlotName="breadcrumbs-slot" />
+        <ExtensionSlot
+          key={pageConfig.slot}
+          extensionSlotName={pageConfig.slot}
+          state={{ basePath }}
+        />
+      </>
+    );
+  };
```

Comment:
Any reason not to just do

```suggestion
    const page = match.params.page;

    return (
      <>
        <ExtensionSlot extensionSlotName="breadcrumbs-slot" />
        <ExtensionSlot
          key={page}
          extensionSlotName={`offline-tools-page-${page}-slot`}
          state={{ basePath }}
        />
      </>
    );
  };
```

? And then get rid of all the `meta` objects? This would simplify the extension API of this module substantially (in addition to the code being simpler and easier to understand).

Reply:
Good observation actually, that would be possible as long as we only require a `page/name` meta from the extensions.

My arguments for using the meta approach are the following:
* I could see more info being provided by the extensions at some point (this will depend on the designs), e.g. a meta value to control the presence/absence of the breadcrumbs, a fixed page title (like in the patient chart, ...). While it's not here right now, it would be very simple to add when the meta approach is already used.
* Consistency with the patient chart. And both "areas" have the same problem to be solved: Providing pages and page links in the sidebar via extensions from foreign modules. I think that being consistent here is good idea for the code base's health ("If you know one, you know the other.").
* Regarding "the code being simpler and easier to understand":
  I think that either approach of how to register/setup the extensions will have to be documented somewhere. I believe that both are a little opaque in their own way. The meta one because it's "heavy", i.e. because you have to setup multiple extensions the right way and your suggestion because it's "magic", i.e. you need to know that the URL segment maps to the extension slot name. In either case a new user would have to scan and understand the code in order to extend it (without docs, that is).

Though I do have to say, maybe this is indeed prematurely optimized. I think in the end I do lean towards using the meta, but only slightly and am fine with using either suggestion. Thoughts are welcome!